api/src/main/java/com/datastrato/gravitino/authorization/Role.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The interface of a role. The role is the entity which has kinds of privileges. One role can have
+ * multiple privileges of one entity. Gravitino chooses to grant one entity to one role to avoid
```

Comment:
Does one role only control one entity? Can it support controlling multiple entities?

Reply:
We can't. If one role bind multiple entities, entities may be from different catalogs. It's hard to map the role to the underlying catalogs. It will bring a complex mapping relation to maintain. For uses, it may be enough clear.

Reply:
I know your concern. If we can have a group, we can add the roles to a group to simplify the operation users avoid creating too many roles.